core/commands/repo.go

```diff
@@ -262,6 +266,89 @@ daemons are running.
 	},
 }
 
+var repoRmRootCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Unlink the root used by the files API.",
+		ShortDescription: `
+'ipfs repo rm-root' will unlink the root used by the files API ('ipfs
+files' commands) without trying to read the root itself.  The root and
+its children will be removed the next time the garbage collector runs,
+unless pinned.
+
+This command is designed to recover form the situation when the root
+becomes unavailable and recovering it (such as recreating it, or
+fetching it from the network) is not possible.  This command should
+only be used as a last resort as using this command could lead to data
+loss if there are unpinned nodes connected to the root.
+
+This command can only run when the ipfs daemon is not running.
+`,
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("confirm", "Really perform operation."),
+		cmdkit.BoolOption("remove-existing-root", "Remove even if it root exists locally."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		confirm, _ := req.Options["confirm"].(bool)
+		removeExistingRoot, _ := req.Options["remove-existing-root"].(bool)
+
+		if !confirm {
+			return fmt.Errorf("this is a potentially dangerous operation please pass --confirm to proceed")
+		}
+
+		configRoot, err := cmdenv.GetConfigRoot(env)
+		if err != nil {
+			return err
+		}
+
+		// Can't use a full node as that interferes with the removal
+		// of the files root, so open the repo directly
+		repo, err := fsrepo.Open(configRoot)
+		if err != nil {
+			return err
+		}
+		defer repo.Close()
+		bs := bstore.NewBlockstore(repo.Datastore())
+
+		// Get the old root and display it to the user so that they can
+		// can do something to prevent from being garbage collected,
+		// such as pin it
+		val, err := repo.Datastore().Get(core.FilesRootKey)
+		if err == ds.ErrNotFound || val == nil {
+			return cmds.EmitOnce(res, &MessageOutput{"Files API root not found.\n"})
```

Comment:
API sounds a bit weird here

```suggestion
			return cmds.EmitOnce(res, &MessageOutput{"FilesRoot not found.\n"})
```